libraries/primitive_consul_lock.rb

```diff
@@ -172,12 +171,60 @@ def to_json
       @h.to_json
     end
 
-    def concurrency
-      @h['concurrency']
-    end
-
     def holders
       @h['holders']
     end
   end
+
+  class SemaphoreNode < Semaphore
+    def self.get_or_create(path)
+      super(path)
+      SemaphoreNode.new(path, @desired_lock)
```

Comment:
here the access to `@desired_lock` is accessing a static attribute on the class.